### PR TITLE
Update filtertable logic and check we have table in initial step

### PR DESCRIPTION
**Pull request by ryanmitchell**
Fixes: https://github.com/tastyigniter/TastyIgniter/issues/744

### CODE REVIEW

File: components/Booking.php [hunk 1]
```diff
@@ -402,7 +402,7 @@ protected function processValidateAfter($validator, $dateTime)
             return $validator->errors()->add('time', lang('igniter.reservation::default.error_invalid_time'));
 
         $autoAllocateTable = (bool)$this->location->getOption('auto_allocate_table', 1);
-        if ($autoAllocateTable AND $this->manager->isFullyBookedOn($dateTime, input('guest')))
+        if ($autoAllocateTable AND ($this->manager->isFullyBookedOn($dateTime, input('guest')) OR !$this->manager->getNextBookableTable($dateTime, input('guest'))))
```
Comment (sampoyigi) [inline components/Booking.php:405]: `isFullyBookedOn ()` method is same as `! getNextBookableTable ()`.. I'll make the changes

**Comment by ryanmitchell** [comment]
Ok. Adding that meant the error appeared on step 1 in calendar mode, whereas it took you through to step 2 without it.

**Comment by sampoyigi** [comment]
Yea but if you look in `isFullyBookedOn ` its basically checking if the result from `getNextBookableTable` is empty.

**Comment by ryanmitchell** [comment]
Yeah I know... couldn't figure out why it made a difference! :) Maybe your refactor means its not needed.

**Comment by sampoyigi** [comment]
Sweet, I'll merge then :)

**Pull request merged**